pkg/upgrade/jiva.go

```diff
@@ -0,0 +1,381 @@
+	//2. Get Jiva Persistent volumes
+	jivaList, err = k.GetPvByCasType([]string{"jiva"}, "")
+	if err != nil {
+		return nil, "", fmt.Errorf("err getting jiva volumes: %s", err.Error())
+	}
+
+	var volumeNames []string
+	var version string
+
+	//3. Write-out names, versions and desired-versions
+	for _, pv := range jivaList.Items {
+		volumeNames = append(volumeNames, pv.Name)
+		if v, ok := jvMap[pv.Name]; ok && len(version) == 0 {
+			version = v.VersionDetails.Status.Current
+		}
+	}
+
+	//4. Check for zero jiva-volumes
+	if len(version) == 0 || len(volumeNames) == 0 {
+		return volumeNames, version, fmt.Errorf("no jiva volumes found")
+	}
+
+	return volumeNames, version, nil
+}
+
+// Returns additional arguments like image-prefix and image-tags
+func addArgs() []string {
+	var result []string
+	if ImagePrefix != "" {
+		result = append(result, fmt.Sprintf("--to-version-image-prefix=%s", ImagePrefix))
+	}
+
+	if ImageTag != "" {
+		result = append(result, fmt.Sprintf("--to-version-image-tag=%s", ImageTag))
+	}
+
+	return result
+}
+
+func yamlToJobSpec(filePath string) (*batchV1.Job, error) {
+	job := batchV1.Job{}
+	// Check if the filepath is a remote-url
+	if strings.HasPrefix(filePath, "http") {
+		res, err := http.Get(filePath)
+		if err != nil {
+			return nil, err
+		}
+		defer res.Body.Close()
+
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return nil, err
+		}
+
+		// unmarshal yaml file into struct
+		err = yaml.Unmarshal(body, &job)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// A file path is given located on local-disk of host
+		yamlFile, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			return nil, err
+		}
+
+		// unmarshal yaml file to structs
+		err = yaml.Unmarshal(yamlFile, &job)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &job, nil
+}
+
+// BuildJivaBatchJob returns Job to be build
+func BuildJivaBatchJob(cfg *jivaUpdateConfig) *batchV1.Job {
+	return NewJob().
+		WithName(cfg.name).
+		WithNamespace(cfg.namespace).
+		WithBackOffLimit(cfg.backOffLimit).
+		WithPodTemplateSpec(
+			func() *core.PodTemplateSpec {
+				return core.NewPodTemplateSpec().
+					WithServiceAccountName(cfg.serviceAccountName).
+					WithContainers(
+						func() *core.Container {
+							return core.NewContainer().
+								WithName("upgrade-jiva-go").
+								WithArgumentsNew(getContainerArguments(cfg)).
+								WithEnvsNew(
+									[]corev1.EnvVar{
+										{
+											Name: "OPENEBS_NAMESPACE",
+											ValueFrom: &corev1.EnvVarSource{
+												FieldRef: &corev1.ObjectFieldSelector{
+													FieldPath: "metadata.namespace",
+												},
+											},
+										},
+									},
+								).
+								WithImage(fmt.Sprintf("openebs/upgrade:%s", cfg.toVersion)).
+								WithImagePullPolicy(corev1.PullIfNotPresent) // Add TTY to openebs/api
+						}(),
+					)
+			}(),
+		).
+		WithRestartPolicy(corev1.RestartPolicyOnFailure). // Add restart policy in openebs/api
```

Comment:
PR open for adding TTY and restart policy in openebs/api: https://github.com/openebs/api/pull/94